Replace setup.sh with npm scripts

fortuna opened the pull request:
This removes the need to have the shell script, having everything managed by `npm` and `package.json`.
The user only needs to call the standard `npm install` and everything will be setup correctly.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/uproxy/uproxy/2609)
<!-- Reviewable:end -->

fortuna commented:
@trevj , let me know what you think about this change.

trevj commented:
I think I like this!

There's a very small bell ringing at the back of my head about using `postinstall` - something about publishing NPMs, which doesn't seem relevant right now.

trevj commented:
BTW, you'll need to update `.travis.yml` before the build will pass - should be trivial.

fortuna commented:
Made some changes. Another look?

- Removed the tsconfig
- Updated all the files mentioning the `setup.sh` script.
- Changed to the `install` script rather than `postinstall`

By the way, you can now use `npm run build` to build, without the need to install grunt globally. `npm test` also works, and we can add more stuff, like `npm run release`.

trevj commented:
:+1:

Yay!
